138673363 Add project id environment variable

lucyconklin opened the pull request.

lucyconklin committed:
Add PROJECT_ID to .env .env.example and spec_helper

[#138673363]

tmikeschu commented on spec/lib/pivotal-tracker/project_spec.rb:
```diff
@@ -27,14 +27,14 @@
 
       it "has the right id" do
-        expect(project.id).to eq(project_id)
+        expect(project.id.to_s).to eq(project_id)
```
@lucyconklin do you know why this coercion is needed after changing to a variable? Just curious.

lucyconklin replied:
It might be better to change to:
expect(project.id).to eq(project_id.to_i)

@tmikeschu thoughts?

tmikeschu replied:
@lucyconklin nevermind! I was confused about how they were different data types, and forgot that it is a string in  `.env`

tmikeschu reviewed (commented):
Thanks, @lucyconklin

jasonnoble commented on spec/lib/pivotal-tracker/project_spec.rb:
```diff
@@ -27,14 +27,14 @@
     let(:project) { subject.find(project_id) }
     context "with a valid project id" do
-      let(:project_id) { 1962995 }
+      let(:project_id) { ENV['PROJECT_ID'] }
```
Let's add a `.to_i` call here.

jasonnoble commented on spec/lib/pivotal-tracker/project_spec.rb:
```diff
@@ -48,7 +48,7 @@
 
   describe ".stories" do
-    let(:project) { subject.find(1962995) }
+    let(:project) { subject.find(ENV['PROJECT_ID']) }
```
Can we use `project_id` here?

tmikeschu replied:
@jasonnoble this is handled in another story (#138673363)

jasonnoble reviewed (changes requested):
Your commit is missing the Tracker story #...

lucyconklin committed:
Add PROJECT_ID environment variable for testing

- It is a string when called, so in the tests it is converted to an integer.

[#138673363]
